src/pyeconet/api.py

```diff
@@ -165,7 +165,7 @@ async def refresh_equipment(self) -> None:
                 _equip_obj: Equipment = None
                 equipment = self._equipment.get(_equip.get("device_name", ""), None)
```

Comment:
```suggestion
                serial = _equip.get("serial_number")
                equipment = self._equipment.get(serial)
```

`device_name` isn't the key for equipment, `serial_number` is. That explains why the below typo never throws for trying to call a non-existent function.